### PR TITLE
AppImage build scripts

### PR DESCRIPTION
**Pull request by TheAssassin**
CC #6112

### CODE REVIEW

File: scripts/linux/build-appimage.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+OLD_CWD=$(readlink -f .)
+
+pushd "$BUILD_DIR"
+
+# standard linuxdeploy pattern
+#see https://docs.appimage.org/packaging-guide/from-source/index.html for more information
+cmake "$REPO_ROOT" -DCMAKE_INSTALL_PREFIX=/usr
+
+make -j$(nproc) VERBOSE=1
+make install DESTDIR=AppDir
```
Comment (janisozaur) [inline scripts/linux/build-appimage.sh]: Please use `ninja` instead

Comment (TheAssassin) [reply scripts/linux/build-appimage.sh]: As long as `ninja` supports the `DESTDIR` part, that should not be possible. I'm quite sure it doesn't, though.

Comment (janisozaur) [reply scripts/linux/build-appimage.sh]: There's no problem using `DESTDIR` with `ninja`.

Comment (TheAssassin) [reply scripts/linux/build-appimage.sh]:
```
+ ninja -j8
[409/409] Linking CXX executable openrct2
+ ninja install DESTDIR=AppDir
ninja: error: unknown target 'DESTDIR=AppDir'
```

Please provide an example how it should be used, or edit the file directly.

Comment (janisozaur) [reply scripts/linux/build-appimage.sh]: You're using it wrong. It's an environment variable, so:
`$ DESTDIR=AppDir ninja install`

**Comment by janisozaur** [comment]
What's the plan here, do you intend to replace our current job producing the binaries with this one or add it as a next one? Is it going to replace just the 64-bit one or 32-bit too?

So far it's missing at least in those areas:
* no `CMAKE_BUILD_TYPE` set
* tests are not run
* the AppImage is not deployed anywhere (I recommend using https://transfer.sh/ so we can test it)

Additional things for consideration:
* verify ccache is used and provide stats used (see https://github.com/OpenRCT2/OpenRCT2/pull/7582 and https://github.com/OpenRCT2/OpenRCT2/pull/7585)
* can AppImage be built for other architectures too? Cross-compiled?
* how do you use `openrct2-cli` with AppImage?
* what's `libcairo2` needed for?

**Comment by TheAssassin** [comment]
I just intended to provide you with a script that bundles OpenRCT2 into an AppImage. It is up to you to integrate this into your existing CI infrastructure. All the script does is build the AppImage, it isn't supposed to become the "main build script". Instead, the script is supposed to be called _after_ all testing has been done.

**Comment by TheAssassin** [comment]
I guess it makes more sense if you pick up my work from here on. I don't really have time to look into how your CI scripts work right now, unfortunately. I invested some extra time back in the days in this project because I wanted to see whether I could solve this backwards compatibility issue mentioned in that issue, however, it turned out to be (almost) impossible to solve.

> can AppImage be built for other architectures too? Cross-compiled?

Sure. Cross-compilation isn't *really* supported yet (as in: we don't have an official guide for it yet -- adding that to my TODO list), but our tools run fine on ARM etc.

> how do you use openrct2-cli with AppImage?

Good question. AppImages are normally "one app = one file", that's our primary use case (see e.g., https://github.com/AppImage/AppImageUpdate/releases).
Some projects have built AppImages that evaluate their own filename to choose the tool to run (e.g., imagemagick; you have to create symlinks with the tool name to that AppImage).
It's up to you to choose which version you prefer.

> what's libcairo2 needed for?

To be honest: I have no clue. appimagetool's AppImage recently started to depend on it. It happens I'm looking into this _right now_. I'll report here as soon as I have more information.

**Comment by IntelOrca** [comment]
@TheAssassin I tried building the app image but I am unable to run it because it can't find the data files, e.g. language files.

The game is programmed to look first in the same directory as the executable but then fallback to `/usr/share/openrct2`. How do required data files for the application work for app images?

```
ted@ted-ubuntu:~/dev$ ./OpenRCT2-x86_64.AppImage
ERROR[../src/openrct2/localisation/LanguagePack.cpp:86 (FromFile)]: Unable to open /home/ted/dev/data/language/en-GB.txt: Unable to open '/home/ted/dev/data/language/en-GB.txt'
ERROR[../src/openrct2/Context.cpp:332 (Initialise)]: Failed to open configured language: Unable to open language 8
ERROR[../src/openrct2/localisation/LanguagePack.cpp:86 (FromFile)]: Unable to open /home/ted/dev/data/language/en-GB.txt: Unable to open '/home/ted/dev/data/language/en-GB.txt'
FATAL[../src/openrct2/Context.cpp:339 (Initialise)]: Failed to open fallback language: Unable to open language 8
```

**Comment by TheAssassin** [comment]
Instead of falling back to `/usr/share/openrct`, fall back to `<path of executable>/../share/openrct2`. That's the modern approach, and works for every prefix.

**Comment by TheAssassin** [comment]
Looks good. By the way, the "other" obvious solution if you wouldn't have wanted to change the code would have been to just `mv usr/share/openrct2/* usr/bin`. But this is definitely the better solution.

**Comment by TheAssassin** [comment]
Does it work? Can I help test some recently built AppImage?

**Comment by IntelOrca** [comment]
@TheAssassin I only built it locally, I haven't set up CI for it yet. We might only build app images for releases rather than every development build. All I do to build it now is:

```
sudo ./scripts/linux/build-appimage-docker.sh
```

If you want it to include discord-rpc then you need to checkout that into the repository before running the script.

**Comment by TheAssassin** [comment]
Built an AppImage, works fine (except for 30 seconds start time, but it seemed to build up some caches... after that it worked fine).

**Comment by TheAssassin** [comment]
Anything that prevents you from merging this PR left? I am waiting for an "official" AppImage.

**Comment by janisozaur** [comment]
@IntelOrca I'd appreciate letting me review the code here, especially when I clearly stated my will to do so.

**Pull request merged**